src/server/oauth2/OAuth2Client.ts

```diff
@@ -174,30 +185,31 @@ export class OAuth2Client {
                 content as string,
             );
 
-            persis.createWithAssociations(
-                {
-                    token: access_token,
-                    expiresAt: expires_in || '',
-                    refreshToken: refresh_token || '',
-                },
-                [
-                    new RocketChatAssociationRecord(
-                        RocketChatAssociationModel.USER,
-                        state,
-                    ),
-                    new RocketChatAssociationRecord(
-                        RocketChatAssociationModel.MISC,
-                        `${this.config.alias}-oauth-connection`,
-                    ),
-                ],
-            );
+            if (!access_token) {
+                return {
+                    status: statusCode,
+                    content: '<div style="display: flex;align-items: center;justify-content: center; height: 100%;">\
+                                <h1 style="text-align: center; font-family: Helvetica Neue;">\
+                                    Oops, something went wrong, please try again or in case it still does not work, contact the administrator.\
+                                </h1>\
+                            </div>',
+                };
+            }
+
+            this.saveToken({
```

Comment:
When calling functions/methods that return promises, we should either `await` or `return` _always_ - here, this function will return the object below before the execution of `saveToken` finishes running